eqasc

aimichal opened the pull request:
@harsh19 Here are my updates to the code at https://github.com/harsh19/Reasoning-Chains-MultihopQA/tree/evaluator

I've left some TODOs in eqasc/README.md and in eqasc/code/README.md. Can you please help fill these out?

aimichal committed:
copy files from harsh19/Reasoning-Chains-MultihopQA (branch evaluator)

aimichal committed:
point to eQASC and TRACIE

aimichal committed:
import json

aimichal committed:
formatting

aimichal committed:
rename arg1

aimichal committed:
take label filename as arg

aimichal committed:
take metrics filename as arg, add logs

aimichal committed:
remove unneeded code (i guess json is provided with one of the * imports)

aimichal committed:
Explain how to run the evaluator in Docker

aimichal committed:
update README

aimichal committed:
update README.md

aimichal committed:
update root README

aimichal commented on eqasc/README.md:
```diff
@@ -0,0 +1,24 @@
+This directory has code and data for the eQASC evaluator, as described in the EMNLP 2020 paper [Learning to Explain: Datasets and Models for Identifying Valid Reasoning Chains in Multihop Question-Answering](https://www.semanticscholar.org/paper/Learning-to-Explain%3A-Datasets-and-Models-for-Valid-Jhamtani-Clark/ebaebfefec9d5c21a4559a1a038743bd437d2f01).
+
+* [code](code/) holds the evaluator
+* [data/](data/) holds the ... TODO
```
These files seem not needed for the evaluator. Can they be removed from this repo?

harsh19 replied:
data/ files ? Yes they can be removed

aimichal replied:
Thanks, I'll move it out.

aimichal commented on eqasc/README.md:
```diff
@@ -0,0 +1,24 @@
+
+* [code](code/) holds the evaluator
+* [data/](data/) holds the ... TODO
+* [evaluator_data/eqasc](evaluator_data/eqasc/) holds the labels used by the evaluator
```
Do you mind if I move these files to a simpler directory name, like `data`?

harsh19 replied:
sure sounds good

aimichal replied:
Alrighty, I'll do this.

aimichal commented on eqasc/code/README.md:
```diff
@@ -0,0 +1,124 @@
+
+This file holds a single JSON structure with three key-value pairs. The keys are:
+
+* `auc_roc` -- This is .. TODO
+* `explainP1` -- This is .. TODO
+* `explainNDCG` - This is ... TODO
```
@harsh19 Please help with these!

harsh19 replied:
Will add

aimichal replied:
Thanks!

harsh19 replied:
Done.

aimichal commented on eqasc/code/README.md:
```diff
@@ -0,0 +1,124 @@
+The chain IDs must match those in the labels. (See below.)
+
+The file `predictions/grc.test.predict` in this repo contains an example
+prediction for the test labels. It was made by doing ... TODO
```
@harsh19 can you explain how this file was made?

harsh19 replied:
Using command at https://github.com/harsh19/Reasoning-Chains-MultihopQA/blob/evaluator/code/scripts/grc.sh#L43

aimichal replied:
Thanks! I'll update the TODO.

aimichal commented on eqasc/code/allennlp_reasoning_explainqa/evaluator/prepare_evaluator_data.py:
```diff
@@ -0,0 +1,63 @@
+
+    print("Counter(all_labels) = ", Counter(all_labels))
+    json.dump(chainid_to_label, open(fwname,'w'))
+    #
+    # mkdir ../evaluator_data/
+    # mkdir ../evaluator_data/eqasc/
+    # env PYTHONPATH=. python allennlp_reasoning_explainqa/evaluator/prepare_evaluator_data.py qasc dev ../evaluator_data/eqasc/chainid_to_label_dev.json
+    # env PYTHONPATH=. python allennlp_reasoning_explainqa/evaluator/prepare_evaluator_data.py qasc test ../evaluator_data/eqasc/chainid_to_label_test.json
```
I think I can nuke these lines

harsh19 replied:
sounds good

aimichal replied:
Alright, I'll do this.

aimichal commented on eqasc/code/environment.yml:
```diff
@@ -0,0 +1,114 @@
+    - zipp==3.1.0
+    - zope-event==4.4
+    - zope-interface==5.1.0
+prefix: /home/jharsh/miniconda3/envs/grc
```
I don't use Conda, but we shouldn't have this line!

harsh19 replied:
Yeah agreed, it shouldn't be there. Will remove it.

aimichal replied:
Since I got the code to use just numpy==1.19.2 and scikit-learn==0.23.2 with Python 3.6.8, your environment.yml file should shrink substantially.

aimichal committed:
remove metrics file, that's generated by running the evaluator

aimichal committed:
Dockerfile

aimichal commented on eqasc/code/scripts/clf.sh:
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
```
Are these scripts needed? They aren't used by the evaluator...

harsh19 replied:
We can remove entire scripts/ folder as none of it is used by evaluator

aimichal replied:
👍 

Thanks! I'll remove it.

aimichal committed:
remove comments in Dockerfile

aimichal committed:
test script

aimichal committed:
remove unused scripts, dirs, and empty modules

aimichal committed:
dead imports

aimichal committed:
more import hygiene

aimichal committed:
comment on origin of grc.test.predict file

aimichal committed:
import json

aimichal committed:
simplify constant

aimichal committed:
delete dataset_reader

aimichal committed:
delete model code

aimichal committed:
remove data code

aimichal committed:
remove dead code in confusion_matrix.py

aimichal committed:
dead code in constants and utils

aimichal committed:
formatting

aimichal committed:
remove comments and dead code

aimichal committed:
more dead code

aimichal committed:
dead var

aimichal committed:
dead function

aimichal committed:
simpler init files

aimichal committed:
redundant parens

aimichal committed:
remove PWC reference

aimichal commented:
@harsh19 I've removed code that wasn't needed by the evaluator, and simplified/formatted the rest. This means diverging from the code you have at https://github.com/harsh19/Reasoning-Chains-MultihopQA which seems reasonable to me, given that this is supposed to be just the evaluator.

I also added a `test-with-docker.sh` script to validate that the evaluator works as expected. Works the same before and after these changes.

Please have a look at the way this came out.

Here's a diff of the `tree .` output, showing that 22 dirs became 8 dirs and 42 files became 17 files:

```diff
--- /tmp/before.txt     2020-11-02 13:59:01.000000000 -0800
+++ /tmp/after.txt      2020-11-02 13:59:14.000000000 -0800
@@ -1,67 +1,28 @@
 .
 ├── README.md
 ├── code
 │   ├── Dockerfile
 │   ├── README.md
 │   ├── allennlp_reasoning_explainqa
 │   │   ├── __init__.py
 │   │   ├── common
 │   │   │   ├── __init__.py
-│   │   │   ├── constants.py
-│   │   │   ├── testing
-│   │   │   │   └── __init__.py
-│   │   │   └── utils.py
-│   │   ├── data
-│   │   │   ├── __init__.py
-│   │   │   ├── analysis
-│   │   │   │   ├── entity_detection.py
-│   │   │   │   └── grc_transformation.py
-│   │   │   ├── dataset_reader
-│   │   │   │   ├── __init__.py
-│   │   │   │   └── qasc_chains_reader.py
-│   │   │   └── fields
-│   │   │       └── __init__.py
+│   │   │   └── constants.py
 │   │   ├── evaluator
 │   │   │   ├── __init__.py
-│   │   │   ├── constants.py
-│   │   │   ├── evaluator.py
-│   │   │   └── prepare_evaluator_data.py
-│   │   ├── model
-│   │   │   ├── __init__.py
-│   │   │   ├── analyze_eqasc_perturbed.py
-│   │   │   ├── bert_chain_clf.py
-│   │   │   ├── interactive.py
-│   │   │   └── retrieval.py
-│   │   ├── tests
-│   │   │   └── __init__.py
+│   │   │   └── evaluator.py
 │   │   └── training
 │   │       ├── __init__.py
 │   │       └── metrics
 │   │           ├── __init__.py
 │   │           ├── confusion_matrix.py
 │   │           └── explanation_eval.py
-│   ├── configs
-│   │   ├── grcfinal.json
-│   │   └── normalfinal.json
 │   ├── environment.yml
 │   ├── predictions
 │   │   └── grc.test.predict
-│   └── scripts
-│       ├── clf.sh
-│       ├── grc.sh
-│       └── retrieval.sh
-├── data
-│   ├── README.md
-│   ├── eobqa
-│   │   └── obqa_chains.tsv.processed.tsv
-│   ├── eqasc
-│   │   └── README.md
-│   ├── eqasc_perturbed
-│   │   └── turk_modified_test.tsv
-│   └── explore_data.ipynb
-└── evaluator_data
-    └── eqasc
-        ├── chainid_to_label_dev.json
-        └── chainid_to_label_test.json
+│   └── test-with-docker.sh
+└── data
+    ├── chainid_to_label_dev.json
+    └── chainid_to_label_test.json
 
-22 directories, 42 files
+8 directories, 17 files
```

aimichal committed:
don't depent on allennlp annotations

aimichal committed:
remove dependency on matplotlib

aimichal committed:
simplify dependencies, no need for allennlp

aimichal commented:
@harsh19 Also, I removed the dependency on AllenNLP in Dockerfile; after some minor tweaks, only scikit-learn and numpy are needed. 👍

harsh19 committed:
Added scores description

harsh19 committed:
Update environment.yml

aimichal committed:
remove dead TODO

aimichal committed:
update wording a bit

aimichal committed:
slashes
